### DISCUSSION TITLE
Allow multiple kernels on one notebook.

### DISCUSSION THREAD

**Discussion by Photonico** [microsoft/vscode-jupyter#9364] [New Ideas]
Hi,

I'm using both Julia and Python for my projects. It is not convenient that only one kernel can be activated in one Jupyter file. 
Optional different kernels in each cell can improve programming performance greatly.

Sincerely
Lu

**Comment by DonJayamanne** [comment]
Out of curiosity do you use Jupyter Lab/Notebooks today? If you do, how do you solve this issue, do you use some package such as [sos](https://vatlab.github.io/sos-docs/), [pyjulia](https://github.com/JuliaPy/pyjulia) or the like?

Also, do you share information betwee the two kernel?
E.g. declare a variable in Julia and access that data in python?
I'm trying to understand how you use both in the same notebook.

FYI - a possible solution is to run Julia code as follows:
```
%%script julia

println("Hello")
```

**Reply by Photonico** [reply]
Hi,

Thanks for your reply.
I only use VSCode Jupyter extension, and I don't know how to active two kernels at the same time.

Sincerely,

**Reply by DonJayamanne** [reply]
Having played with [pyjulia](https://pyjulia.readthedocs.io/en/latest/usage.html) a little, I find this the most promising and easiest of all options when playing with the two languages together in the same notebook. However my usage has only be limited to explorations and the like.

**Reply by DonJayamanne** [reply]
@Photonico in that case, for now I'd suggest having a look at either [pyjulia](https://pyjulia.readthedocs.io/en/latest/usage.html) or just running julia using the following syntax (you don't need to install anything, just select a Python kernel & it will just work):
```
%%script julia

println("Hello")
```

**Reply by Photonico** [reply]
Thanks a lot! it works, but the Syntax highlighting hint is still based on Python.

**Reply by Photonico** [reply]
I prefer this feature to be a native of Jupyter extension.

**Reply by DonJayamanne** [reply]
> I prefer this feature to be a native of Jupyter extension.

Thanks for your interest and suggestion, unfortunately the Jupyter extension in VS Code will only provide functionality thats already available in Jupyter Notebooks/Lab, thats because this extension tries to maintain parity with the Jupyter application within VS code. However that doesn't mean anyone else cannot create such an extension.

As for the syntax highlighting, you can follow up on that issue here https://github.com/microsoft/vscode-jupyter/issues/4588

**Reply by Photonico** [reply]
I found that clicking the lower right button of the cell can demonstrate the menu which controls the syntax highlighting. 

If Julia option would be added to the menu, the problem gets solved.
![pic](https://raw.githubusercontent.com/Photonico/Problems/master/2022/Screenshot_20220316_062224.png)

**Comment by claudiaregio** [comment]
@Photonico, I work on [.NET Interactive](https://github.com/dotnet/interactive) which enables a polyglot experience in Jupyter notebooks. We are working on a solution that would also support Julia, I'd love to chat more with you about your use cases and expectations. 

@davidanthoff

**Reply by Photonico** [reply]
That's great! 

My expectation is that we can demonstrate both Julia, Python, and Markdown in one file. 
We can write the same algorithm in both Julia and Python to show the differences between these languages in an intuitive way.

Thanks

**Comment by gjsjohnmurray** [comment]
Now that .NET Interactive has [hit the buffers](https://github.com/dotnet/interactive/issues/4163) perhaps this idea can be reassessed here.
